### PR TITLE
Add Azure and GCP support for ObjectStore Cloud backend 

### PR DESCRIPTION
**Pull request by VJalili**

### CODE REVIEW

File: lib/galaxy/objectstore/cloud.py [hunk 1]
```diff
@@ -92,15 +89,121 @@ def _initialize(self):
...
+
+            if len(missing_config) > 0:
+                msg = "The following configuration required for {} cloud backend " \
+                      "are missing: {}".format(provider, missing_config)
+                log.error(msg)
+                raise Exception(msg)
+            else:
+                return config
+        except Exception:
+            log.exception("TMalformed ObjectStore Configuration XML -- unable to continue")
```
Comment (jmchilton) [inline lib/galaxy/objectstore/cloud.py]: TMalformed

**Comment by jmchilton** [comment]
Can you add some more serialization tests for the new stuff (https://github.com/galaxyproject/galaxy/blob/dev/test/unit/test_objectstore.py#L543), test coverage for configuring these things is getting better I think we should not backtrack here.

**Comment by VJalili** [comment]
Agree, I am working on the tests now. First updating to adhere with new changes, and will then extend to assert new vendors.

**Review by jmchilton — APPROVED** [review]

File: config/object_store_conf.xml.sample [hunk 2]
```diff
@@ -46,22 +46,44 @@
...
+        -->
+
+        <!-- Cloud ObjectStore: Google Compute Engine (GCE) -->
+        <!--
+        <object_store type="cloud" provider="google" order="0">
+            <auth credentials_file="..." />
+            <bucket name="..." use_reduced_redundancy="False" />
+            <cache path="database/object_store_cache" size="1000" /> <!-- size is in gigabytes -->
+            <extra_dir type="job_work" path="database/job_working_directory_gce"/>
+            <extra_dir type="temp" path="database/tmp_gce"/>
```
Comment (afgane) [inline config/object_store_conf.xml.sample]: Perhaps better to go with the Google Cloud **Platform** (GCP) terminology vs. GCE so it's consistent with CloudBridge (and Google). xref https://github.com/CloudVE/cloudbridge/pull/189

**Review by afgane — APPROVED** [review]
LGTM with just the suggestion or renaming GCE to GCP throughout the PR.

File: config/object_store_conf.xml.sample (hunk 2 above)
Comment (VJalili) [reply config/object_store_conf.xml.sample]: Done.

**Comment by VJalili** [comment]
@afgane Thank!

**Pull request merged**